lib/request.js

```diff
@@ -233,11 +242,25 @@ Request.prototype.sendResponse = function(response) {
 
   response.writeHead(this.response.statusCode, headers);
 
-  if (this.response.filePath) {
-    var readStream = fs.createReadStream(this.response.filePath);
-    readStream.pipe(response);
+  if (isStream(this.response.body)) {
+    // Because we may need to respond with this body more than once, if it is a stream,
+    // we make a buffer copy and use that as the body for future responses.
```

Comment:
The flip side is we already keep a copy of the `body` for any other requests where `_maxRequests > 1`, so in theory this problem already exists for arbitrarily sized bodies.

Maybe we should just take this as is. <ponders>